### PR TITLE
Add user local data preprocess support

### PR DESCRIPTION
**Pull request by lidanqing-intel**
To provide demo to users, we need to explain how to preprocess and convert data to binary. There should be an option to choose users' local data.
Added unit test for it.

@wojtuss Please review, thank you

### CODE REVIEW

File: paddle/fluid/inference/tests/api/full_ILSVRC2012_val_preprocess.py [hunk 1]
```diff
@@ -205,9 +206,86 @@ def run_convert():
...
+                of.write(np_img.astype('float32').tobytes())
+
+                #save label(int64_t) to file
+                label_int = (int)(label)
+                np_label = np.array(label_int)
+                of.seek(SIZE_INT64 + SIZE_FLOAT32 * DATA_DIM * DATA_DIM * 3 *
+                        num_images + idx * SIZE_INT64)
+                of.write(np_label.astype('int64').tobytes())
+
+        target_size = 8 + num_images * 3 * args.data_dim * args.data_dim * 4 + num_images * 8
```
Comment (wojtuss) [inline paddle/fluid/inference/tests/api/full_ILSVRC2012_val_preprocess.py]: Could you please add an explanation of what particular factors mean?

File: paddle/fluid/inference/tests/api/full_ILSVRC2012_val_preprocess.py [hunk 2]
```diff
@@ -205,9 +206,86 @@ def run_convert():
...
+        else:
+            print("Conversion failed!")
+
+
+def main_preprocess_Imagenet(args):
+    parser = argparse.ArgumentParser(
+        description="Convert the full Imagenet val set or local data to binary file.",
+        usage=None,
+        add_help=True)
+    parser.add_argument('--local', action="store_true")
```
Comment (wojtuss) [inline paddle/fluid/inference/tests/api/full_ILSVRC2012_val_preprocess.py]: Please, add a description of this option.

File: paddle/fluid/inference/tests/api/full_ILSVRC2012_val_preprocess.py [hunk 3]
```diff
@@ -205,9 +206,86 @@ def run_convert():
...
+    parser.add_argument(
+        "--data_dir",
+        default="./third_party/inference_demo/int8v2/imagenet_small",
```
Comment (wojtuss) [inline paddle/fluid/inference/tests/api/full_ILSVRC2012_val_preprocess.py]: I think I would not define the default values for this path and the next filename/path options. These values would be set either in the test definition (in a cmake file) or deliberately by a user. In both cases setting the default values here can be misleading.

File: paddle/fluid/inference/tests/api/full_pascalvoc_test_preprocess.py [hunk 4]
```diff
@@ -268,8 +268,7 @@ def main_pascalvoc_preprocess(args):
         description="Convert the full pascalvoc val set or local data to binary file.",
         usage=None,
         add_help=True)
-    parser.add_argument(
-        '--choice', choices=['local', 'VOC_test_2007'], required=True)
+    parser.add_argument('--local', action="store_true")
```
Comment (wojtuss) [inline paddle/fluid/inference/tests/api/full_pascalvoc_test_preprocess.py]: Same as above.

File: paddle/fluid/inference/tests/api/full_ILSVRC2012_val_preprocess.py (hunk 3 above)
Comment (lidanqing-vv) [reply paddle/fluid/inference/tests/api/full_ILSVRC2012_val_preprocess.py]: You are right. I will change this.

File: paddle/fluid/inference/tests/api/full_pascalvoc_test_preprocess.py (hunk 4 above)
Comment (lidanqing-vv) [reply paddle/fluid/inference/tests/api/full_pascalvoc_test_preprocess.py]: Done

File: paddle/fluid/inference/tests/api/full_ILSVRC2012_val_preprocess.py (hunk 3 above)
Comment (lidanqing-vv) [reply paddle/fluid/inference/tests/api/full_ILSVRC2012_val_preprocess.py]: Done

File: paddle/fluid/inference/tests/api/full_ILSVRC2012_val_preprocess.py (hunk 2 above)
Comment (lidanqing-vv) [reply paddle/fluid/inference/tests/api/full_ILSVRC2012_val_preprocess.py]: Done

**Pull request merged**